### DISCUSSION TITLE
APCO

### DISCUSSION THREAD

**Discussion by tinkeringtuck** [tronikos/opower#27] [General]
It looks like Appalachian Power Company (APCO - an AEP company), is using opower so I am started a draft PR and copied one of the utility py's to take a stab. Looking briefly, it seems like there are other AEP companies so it may make more sense long term to take the Exelon approach but not sure. I'm going to give a stab and see if I can reverse engineer what's been setup already to get it to work for APCO but definitely open to help/pointers/contribution.

**Comment by tronikos** [comment]
I took a quick look. All the seven AEP subsidiaries seem to use similar sites so if you can figure out what opower subdomain each one is using, it does make sense to take the same approach I took with Exelon.

**Reply by tinkeringtuck** [reply]
sounds good, I'll keep that in mind. I've started by trying to reverse engineer your examples by using tools like "developer tools" in my browser and postman starting with the login url and seeing if I can understand how you knew to select certain parts i.e. how you found the verification token for the handle_starttag. Do you have any tips for doing this sort of reverse engineering or any tools in particular that are useful? Thanks, I'm making progress but it's slow and a lot of guess and check.

**Comment by tronikos** [comment]
Yes it's a tedious work with a lot of trial error. I've only used the browser developer tools.

**Reply by tinkeringtuck** [reply]
Well, one benefit is I am learning a lot about how to use developer tools!
